common/server_player.cpp

```diff
@@ -797,6 +799,24 @@ Server_Player::cmdUnconcede(const Command_Unconcede & /*cmd*/, ResponseContainer
     return Response::RespOk;
 }
 
+Response::ResponseCode Server_Player::cmdJudge(const Command_Judge &cmd, ResponseContainer &rc, GameEventStorage &ges)
+{
+    if (!judge)
+        return Response::RespFunctionNotAllowed;
+
+    Server_Player *player = this->game->getPlayers()[cmd.target_id()];
+
+    ges.setForcedByJudge(playerId);
+    if (player == nullptr)
+        return Response::RespContextError;
+
+    for (int i = 0; i < cmd.game_command_size(); ++i) {
+        player->processGameCommand(cmd.game_command(i), rc, ges);
```

Comment:
Not really a change request, but I was a bit in doubt about this; maybe you already checked it, but I thought it could be useful to point this out.

When servatrice runs multithreaded, multiple connection pools can live in different threads, so command sent by 2 different players on the same game can be handled in 2 different instances of `Server_ProtocolHandler` living in different threads.
Anyway, it seems like `Server_Game::addPlayer()` already takes care of this by ensuring that every `Server_Player` object lives in the same thread of the game.